File: Platform/Platform.Models/WeatherForecast.cs
using System;
using Platform.Models.Attributes;
using Platform.Models.Enums;
using Platform.Models.Interfaces;

namespace Platform.Models
{
    public class WeatherForecast : IPlatformModel
    {
        public long Id { get; set; }

        [PlatformAttribute(AttributesEnum.Grid | AttributesEnum.Form)]
        public DateTime Date { get; set; }

        [PlatformAttribute(AttributesEnum.Grid)]
        public int TemperatureC { get; set; }

        [PlatformAttribute(AttributesEnum.Form)]
        public int TemperatureF => 32 + (int)(TemperatureC / 0.5556);

        public string Summary { get; set; }

        [PlatformAttribute(AttributesEnum.Form)]
        public int MyProperty { get; set; }

        public CurrencyReferenceBook Currency { get; set; }
    }
}

File: Platform/Platform.Fodels/Models/WeatherForecast.fs
namespace Platform.Fodels.Models

open System;
open Platform.Fodels.Interfaces
open Platform.Fodels.Enums
open Platform.Fodels.Attributes

type [<AllowNullLiteral>] WeatherForecast() =
    interface IPlatformModel with
        member this.Id
            with get () = this.id
            and set (value) = this.id <- value

    [<DefaultValue>]
    val mutable private id: int

    [<DefaultValue>]
    val mutable private date: DateTime
    
    [<Platform(AttributesEnum.Form ||| AttributesEnum.Grid)>]
    member this.Date
        with get () = this.date
        and set (value) = this.date <- value

    [<DefaultValue>]
    val mutable private tempC: int
    
    [<Platform(AttributesEnum.Grid)>]
    member this.TemperatureC
        with get () = this.tempC
        and set (value) = this.tempC <- value

    [<DefaultValue>]
    val mutable private tempF: int
    member this.TemperatureF
        with get () = this.tempF
        and set (value) = this.tempF <- value

    [<DefaultValue>]
    val mutable private myProp: int
    
    [<Platform(AttributesEnum.Form)>]
    member this.MyProperty
        with get () = this.myProp
        and set (value) = this.myProp <- value

    [<DefaultValue>]
    val mutable private sum: string
    member public this.Summary
        with get () = this.sum
        and set value = this.sum <- value

